A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/CDU/A320_Neo_CDU_PerformancePage.js

```diff
@@ -558,18 +558,25 @@ class CDUPerformancePage {
         mcdu.onRightInput[1] = () => {
             const value = mcdu.inOut;
             mcdu.clearUserInput();
-            if (mcdu.setPerfApprMDA(value)) {
+            if (mcdu.setPerfApprMDA(value) && mcdu.setPerfApprDH(FMCMainDisplay.clrValue)) {
                 CDUPerformancePage.ShowAPPRPage(mcdu);
             }
         };
         let dhCell = "[ ]";
         if (isFinite(mcdu.perfApprDH)) {
-            dhCell = mcdu.perfApprDH.toFixed(0);
+            if (mcdu.perfApprDH >= 0 && mcdu.perfApprDH <= 700) {
```

Comment:
I think this check can be moved in the `setPerfApprDH(s)` function, since you want to check the values on input instead of on display.

Reply:
Fully agree 👍
Check moved to `setPerfApprDH(s)`
Thank you